src/main/java/com/bina/varsim/tools/simulation/VCF2diploid.java

```diff
@@ -361,24 +361,33 @@ private boolean addVariant(final byte[] maskedSequence, final Sequence reference
         variants in the same haplotype should never overlap.
         Otherwise, they should be merged at first place.
          */
-        for (int p = position; p < position + referenceAlleleLength; p++) {
-            // if any location of this variant overlap a deleted base or a SNP, we skip it
-            // DELETED_BASE is used as a flag to mark that this position has been processed/modified
-            if (maskedSequence[p - 1] == DELETED_BASE
-                    // insertions may not be next to SNPs or deletions, this is to avoid problem of DUP/INV
-                    //why p - 1 for maskedSequence? is it 1-based or 0-based?
-                    || maskedSequence[p - 1] != referenceSequence.byteAt(p)) {
-                overlap = true;
+        if (referenceAlleleLength == 0) {//insertion or duplication
+            // insertions may not be surrounded by deletions
+                if (maskedSequence[position - 1] == DELETED_BASE &&
+                        (position >= 2 && maskedSequence[position - 2] == DELETED_BASE) &&
```

Comment:
Why are you looking at `position 2`?

Reply:
`position` is 1-based, while `maskedSequence` is 0-based. `position - 2` is the bp before insertion locus, `position - 1` is the bp after insertion locus. 

actually there is no need to check `position`, which is 2bp after insertion locus. I just removed it. Thanks.